File: anthropic-cli-chatbot/node/chatbot.js
import Anthropic from "@anthropic-ai/sdk";
import dotenv from "dotenv";
import readline from "readline";

dotenv.config();

const client = new Anthropic();
const messages = [];

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const ask = (q) => new Promise((resolve) => rl.question(q, resolve));

async function chat() {
  console.log("Claude Chatbot (type 'exit' to quit)\n");

  while (true) {
    const input = await ask("You: ");
    if (input.toLowerCase() === "exit") break;

    messages.push({ role: "user", content: input });

    try {
      const res = await client.messages.create({
        model: "claude-sonnet-4-6",
        max_tokens: 1024,
        messages,
      });

      const reply = res.content[0].text;
      messages.push({ role: "assistant", content: reply });
      console.log("Bot:", reply, "\n");
    } catch (err) {
      console.error("Error:", err.message, "\n");
    }
  }

  rl.close();
  console.log("Goodbye!");
}

chat();

File: anthropic-cli-chatbot/python/chatbot.py
import os
from openai import OpenAI
from dotenv import load_dotenv

load_dotenv(override=True)

client = OpenAI(
    base_url="https://integrate.api.nvidia.com/v1",
    api_key=os.environ["NVIDIA_API_KEY"],
)

messages = []

print("NIM Chatbot — Llama 3.1 8B (type 'exit' to quit)\n")

while True:
    user_input = input("You: ")
    if user_input.lower() == "exit":
        break

    messages.append({"role": "user", "content": user_input})

    try:
        response = client.chat.completions.create(
            model="meta/llama-3.1-8b-instruct",
            messages=messages,
        )
        reply = response.choices[0].message.content
        messages.append({"role": "assistant", "content": reply})
        print(f"Bot: {reply}\n")
    except Exception as e:
        print(f"Error: {e}\n")

print("Goodbye!")
